pgcli/pgexecute.py

```diff
@@ -215,6 +221,9 @@ def connect(self, database=None, user=None, password=None, host=None,
         self.host = host
         self.port = port
 
+        if not self.host:
+            self.host = self.socket_directory()
```

Comment:
If you assign socket directory to `self.host`, it makes `self.host` mean different things at different times. It may also interfere with the current behavior if you mess with `self.host` - possibly affect reconnecting on timeout. I think it's better if we have no host assigned when connected via socket.